script/bin/operator.rs

```diff
@@ -281,33 +280,126 @@ fn get_block_update_interval() -> u64 {
+        const DEFAULT_PATH: &str = "chains.json";
+
+        let path = env::var("CHAINS_PATH").unwrap_or(DEFAULT_PATH.to_string());
+
+        Self::from_file(&path).or_else(|_| {
+            tracing::info!("No chains file found, trying env.");
+            Self::from_env()
+        })
+    }
+
+    /// Tries to read from the `CHAINS` environment variable.
+    fn from_env() -> Result<Vec<Self>> {
+        let chains = env::var("CHAINS")?;
+
+        Ok(serde_json::from_str(&chains)?)
+    }
+
+    fn from_file(path: &str) -> Result<Vec<Self>> {
+        tracing::debug!("Reading chains from file: {}", path);
+
+        let file = std::fs::read_to_string(path)?;
+
+        Ok(serde_json::from_str(&file)?)
+    }
+}
+
 #[tokio::main]
 async fn main() {
     dotenv::dotenv().ok();
-    env_logger::init();
-
-    let operator = SP1BlobstreamOperator::new().await;
-
-    info!("Starting SP1 Blobstream operator");
-    const LOOP_TIMEOUT_MINS: u64 = 20;
-    loop {
-        let request_interval_mins = get_loop_interval_mins();
-        tokio::select! {
-            _ = tokio::time::sleep(tokio::time::Duration::from_secs(60 * LOOP_TIMEOUT_MINS)) => {
-                log::error!("Operator took longer than {} minutes to run.", LOOP_TIMEOUT_MINS);
-                continue;
-            }
-            e = operator.run() => {
-                if let Err(e) = e {
-                    // Sleep for less time on errors.
-                    error!("Error running operator: {:?}", e);
 
-                    tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
-                    continue;
+    // Setup tracing.
+    tracing_subscriber::fmt::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    // Succinct deployments use the `CHAINS` environment variable.
+    let config = ChainConfig::fetch().expect("Failed to fetch chain config");
+    let maybe_private_key: Option<PrivateKeySigner> = env::var("PRIVATE_KEY")
+        .ok()
+        .map(|s| s.parse().expect("Failed to parse PRIVATE_KEY"));
+
+    // Setup the KMS relayer config.
+    let use_kms_relayer: bool = env::var("USE_KMS_RELAYER")
+        .map(|s| s.parse().expect("USE_KMS_RELAYER failed to parse"))
+        .expect("USE_KMS_RELAYER not set");
+
+    // Ensure we have a signer if we're not using the KMS relayer.
+    if !use_kms_relayer && maybe_private_key.is_none() {
+        panic!("PRIVATE_KEY is not set but USE_KMS_RELAYER is false.");
+    }
+
+    // Setup our signer.
+    let signer = MaybeWallet::new(maybe_private_key.map(EthereumWallet::new));
+
+    // Setup the prover and program keys.
+    let prover = ProverClient::builder().cpu().build();
+    let (pk, vk) = prover.setup(TENDERMINT_ELF);
+    let pk = Arc::new(pk);
+
+    // Setup all the tasks.
+    // These futures should never resolve, so we just await them in the main thread.
+    let handles = config.into_iter().map(
+        |c| {
+            let provider = ProviderBuilder::new()
```

Comment:
Here, you'd look at all deployments. See what the latest block is.

Then for each unique latest block, do the input fetching and request the proofs. Once the proofs come back, then relay on-chain.

In happy path, same proof is used.